### PR TITLE
feat(models): Add ModelInterface

### PR DESCRIPTION
**Pull request by matschreiner**
# Decouple model from task 
                                                                                                             
#  Summary     
                                                                                                             
  This PR adds an interface for AnemoiModels 

#  Motivation                                                                                                 
The old AnemoiModelInterface was a fatal misnomer that gave the impression of having some decoupling between models and task, but really it was responsible for building processors, instantiating the backbone, holding metadata and supporting arrays, and being constructed inside the Lightning task.

# Changes                                                                                                    
anemoi-models                                                                                              
  - AnemoiModelInterface replaced by ModelInterface — a @runtime_checkable Protocol defining the inference
  surface (pre_process, post_process, forward, predict_step). No Lightning dependency.                       
  - New AnemoiModel (models/anemoi_model.py) — concrete torch.nn.Module implementing ModelInterface. Holds
  backbone, pre/post processors, graph data and data indices. Pre/post processing is now model.pre_process(x)
   / model.post_process(y).                                                                                  
  - New NaiveModel — simple passthrough model for debugging/baselines.
  - Pre/post process steps removed from individual backbone classes (encoder_processor_decoder.py,           
  interpolator.py).                                                                                          
                                                                                                             
anemoi-training                                                                                            
  - New builder.py — factory function build_anemoi_model(...) called via instantiate(config.model) in        
  train.py. Builds processors, graph, supporting arrays, metadata, and returns a fully constructed           
  AnemoiModel.                                                                                    
  - All tasks (BaseGraphModule and subclasses) now accept a pre-built model: ModelInterface instead of       
  constructing it internally.                                                                         
  - Pre-processing moved from _normalize_batch into on_before_batch_transfer via model.pre_process(batch).   
  - Post-processing for validation metrics now explicit via model.post_process(y) in _step.               
  - Metadata and supporting arrays moved from the model to the task (self.metadata, self.supporting_arrays), 
  and checkpoint callbacks updated accordingly.                                                              
  - Supporting array construction (including output masks) moved to utils/supporting_arrays.py.              
  - Node name in graph access no longer hardcoded — uses config.graph.data.                                  
                                                                                                             
#  Known limitations                                                                                                                                                                                     
- Diffusion tasks (BaseDiffusionForecaster, GraphDiffusionTendForecaster) still access self.model.backbone directly for diffusion-specific methods (fwd_with_preconditioning, sigma_max/min, compute_tendency, etc.)

### CODE REVIEW

File: models/src/anemoi/models/interface/__init__.py [hunk 1]
```diff
@@ -7,197 +7,36 @@
...
-        statistics: dict,
-    ) -> tuple[Processors, Processors]:
-        processors = [
-            [name, instantiate(processor, data_indices=data_indices, statistics=statistics)]
-            for name, processor in processors_configs.items()
-        ]
-        return Processors(processors), Processors(processors, inverse=True)
-
-    def _build_tendency_processors(
+
```
Comment (matschreiner) [inline models/src/anemoi/models/interface/__init__.py:16]: Remove all functionality from AnemoiModelInterface. Move construction logic into builder.py, move model-logic to anemoi_model.py and make interface a pure interface

File: models/src/anemoi/models/models/base.py [hunk 2]
```diff
@@ -332,8 +334,3 @@ def predict_step(
                     y_hat[dataset_name] = gather_tensor(y_hat[dataset_name], -2, y_hat_shard_shapes, model_comm_group)
 
         return y_hat
-
-    @abstractmethod
-    def fill_metadata(self, md_dict) -> None:
-        """To be implemented in subclasses to fill model-specific metadata."""
-        pass
```
Comment (matschreiner) [inline models/src/anemoi/models/models/base.py:339]: moved to basetask

File: models/src/anemoi/models/schemas/models.py [hunk 3]
```diff
@@ -206,12 +210,14 @@ class DiffusionSchema(BaseModel):
...
     "Feature tensor size in the hidden space."
     keep_batch_sharded: bool = Field(default=True)
     "Keep the input batch and the output of the model sharded"
-    model: Model = Field(default_factory=Model)
-    "Model schema."
+    backbone: Optional[Model] = Field(default=None)
+    "Backbone model schema."
     trainable_parameters: TrainableParameters = Field(default_factory=TrainableParameters)
     "Learnable node and edge parameters."
     bounding: list[Bounding]
```
Comment (matschreiner) [inline models/src/anemoi/models/schemas/models.py:228]: should this schema be so strict?

File: training/src/anemoi/training/diagnostics/callbacks/checkpoint.py [hunk 4]
```diff
@@ -142,12 +136,12 @@ def on_train_start(self, trainer: pl.Trainer, pl_module: pl.LightningModule) ->
         del pl_module
 
         if trainer.is_global_zero:
-            model = self._torch_drop_down(trainer)
+            model = trainer.lightning_module.model
```
Comment (matschreiner) [inline training/src/anemoi/training/diagnostics/callbacks/checkpoint.py:139]: not necessary with torch_drop_down, pl exposes the unwrapped DDP module via trainer.lightning_module directly

File: training/src/anemoi/training/train/train.py [hunk 5]
```diff
@@ -242,34 +262,30 @@ def _validate_transfer_learning_datasets(
...
-                and ".Transformer" in self.config.model.decoder.target_
-            )
-        ), "GLU activation function is not supported in Transformer models, due to fixed dimensions. "
-        "Please use a different activation function."
+        model_task = get_class(self.config.training.model_task)
+
+        # Sync multistep values from training to model config. Interpolations like
+        # model.multistep_input = ${training.multistep_input} may have been frozen by an
+        # earlier OmegaConf.resolve() call before the training values were modified.
+        with open_dict(self.config.model):
```
Comment (matschreiner) [inline training/src/anemoi/training/train/train.py]: figure out what to do here.. Too ugly

File: models/src/anemoi/models/models/base.py [hunk 6]
```diff
@@ -63,29 +63,31 @@ def __init__(
...
         self._assert_matching_indices(data_indices)
-        self._assert_hidden_nodes_name(self._graph_name_hidden)
+        self._assert_hidden_nodes_name(self.hidden_nodes_name)
 
         # build networks
         self._build_networks(model_config)
 
         # build residual connection
-        self._build_residual(model_config.model.residual)
+        self._build_residual(model_config.residual)
```
Comment (matschreiner) [inline models/src/anemoi/models/models/base.py:90]: should this be in the modelbuilder?

File: training/src/anemoi/training/diagnostics/plots.py [hunk 7]
```diff
@@ -10,6 +10,10 @@
 
 import logging
 
+import matplotlib as mpl
```
Comment (matschreiner) [inline training/src/anemoi/training/diagnostics/plots.py]: why do we need this?
